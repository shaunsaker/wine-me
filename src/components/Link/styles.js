import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {},
  text: {
    ...styleConstants.fonts.types.heading,
    color: styleConstants.colors.secondary,
  },
});

export default styles;
