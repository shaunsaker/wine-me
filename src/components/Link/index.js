import React from 'react';
import PropTypes from 'prop-types';
import { Text } from 'react-native';

import styles from './styles';

import Touchable from '../Touchable';

const propTypes = {
  handlePress: PropTypes.func,
  testID: PropTypes.string,
  text: PropTypes.string,
  disabled: PropTypes.bool,
};

const defaultProps = {};

const Link = ({ handlePress, testID, text, disabled }) => {
  return (
    <Touchable onPress={handlePress} style={styles.container} disabled={disabled} testID={testID}>
      <Text style={styles.text}>{text}</Text>
    </Touchable>
  );
};

Link.propTypes = propTypes;
Link.defaultProps = defaultProps;

export default Link;
