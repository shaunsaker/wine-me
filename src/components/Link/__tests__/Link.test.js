import React from 'react';
import renderer from 'react-test-renderer';

import Link from '..';

describe('Link', () => {
  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<Link />);

      expect(component).toMatchSnapshot();
    });
  });
});
