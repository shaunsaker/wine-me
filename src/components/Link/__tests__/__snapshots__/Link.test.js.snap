// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders renders with minimum required props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#3D84A8",
        "fontFamily": "Nunito-SemiBold",
        "fontSize": 17,
      }
    }
  />
</View>
`;
